### PR TITLE
js-pre-oop

### PR DESCRIPTION
**Pull request by merowing**
### js-pre-oop

[Demo](https://merowing.github.io/a-tiny-JS-world/) |
[Code base](https://github.com/merowing/a-tiny-JS-world)

This is the tiny JS World.

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by merowing** [comment]
What I did:

- used prettier
- used local live-server for check the code

File: submissions/merowing/js-pre-oop/index.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+      name: 'Anne',
+      gender: 'female',
+      species: 'human animal',
+      friends: [0, 1, 2],
+   }
+};
+data.catWoman.say = data.cat.say;
+data.woman.species = data.man.species;
+
+const sequences = ['species', 'name', 'gender', 'hands', 'legs', 'say', 'friends'];
```
Comment (OleksiyRudenko) [inline submissions/merowing/js-pre-oop/index.js:60]: Devise a better naming. What does the variable actually contain?

File: submissions/merowing/js-pre-oop/index.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+creatures.forEach(value => {
+   let creature = data[value];
+
+   let message = sequences.map(creatureProperty => {
+      let value = '';
+      
+      switch(creatureProperty) {
+         case 'name':
+            value = `<strong>${creature[creatureProperty]}</strong>`;
+            break;
```
Comment (OleksiyRudenko) [inline submissions/merowing/js-pre-oop/index.js:72]: `switch` is an impretaive way to write code.
Consider using a dictionary (object literal) to define modifiers for properties. Then building an output could be as short as ``.map(propertyName => `<${tags[propertyName]}>${creature[propertyName]}</${tags[propertyName]}>` )``
This way the code would be more declarative, i.e. easier to read and modify.

P.S. There are "null" tags that you could benefit from.

Also consider splitting the handling of properties into a chain of transformations.
E.g. "Remove values that are 'empty', convert arrays into strings, format strings".
Having multiple simple operations vs one complex operation adds to readability and maintainability of code.

**Review by OleksiyRudenko — APPROVED** [review]
@merowing well done!
Check comments below for ideas to absorb and use when solving coding problems.

**Pull request merged**